src/main/java/spoon/support/reflect/code/CtInvocationImpl.java

```diff
@@ -175,6 +176,8 @@ public void replace(CtStatement element) {
 		if (type != null) {
 			type.setParent(this);
 		}
+		// even if the setter delegates to getExecutable, it makes sense to create an event
+		getFactory().getEnvironment().getModelChangeListener().onObjectUpdate(this, TYPE, type, getExecutable().getType());
```

Comment:
So model change listener will be called twice for single modification? I do not like it. 
@monperrus Could you explain why it is good?

Reply:
you're right, it's not good.